dali/kernels/signal/fft/stft_gpu_test.cu

```diff
@@ -0,0 +1,328 @@
+  for (int i = 0; i < out.size(); i++)
+    out[i] = in[i];
+}
+
+template <typename T>
+void RefSpectrum(const OutTensorCPU<T, 2> &ref_out, const float *in, int n,
+                 const StftArgs &args, span<const float> window) {
+  int t_axis = args.time_major_layout ? 0 : 1;
+  int f_axis = 1 - t_axis;
+  int nout = args.num_windows(n);
+  int nfft = args.nfft > 0 ? args.nfft : args.window_length;
+  test::ReferenceFFT<float> fft(nfft);
+  vector<float> ref_wnd(nfft);
+  vector<complexf> ref_fft(nfft);
+  assert(window.size() <= nfft);
+  int out_nfft = ref_out.shape[f_axis];
+  assert(out_nfft <= nfft);
+  vector<T> ref(out_nfft);
+
+  for (int i = 0; i < nout; i++) {
+    RefExtractWindow(ref_wnd.data(), i, in, n, args);
+    for (int j = 0; j < window.size(); j++)
+      ref_wnd[j] *= window[j];
+
+    fft(ref_fft.data(), ref_wnd.data());
+    RefPostprocess(make_span(ref), make_span(ref_fft), args.spectrum_type);
+
+    int pos[2] = {};
+    pos[t_axis] = i;
+    for (int j = 0; j < out_nfft; j++) {
+      pos[f_axis] = j;
+      *ref_out(pos) = ref[j];
+    }
+  }
+}
+
+template <typename T>
+void RefSpectrum(const OutListCPU<T, 2> &ref_out, const InListCPU<float, 1> &in,
+                 const StftArgs &args, span<const float> window) {
+  for (int i = 0; i < in.num_samples(); i++) {
+    RefSpectrum(ref_out[i], in.data[i], in.shape[i][0], args, window);
+  }
+}
+
+template <typename T>
+void RefSpectrum(TestTensorList<T, 2> &ref_out, const InListCPU<float, 1> &in,
+                 const StftArgs &args, span<const float> window) {
+  int t_axis = args.time_major_layout ? 0 : 1;
+  int f_axis = 1 - t_axis;
+  TensorListShape<2> ref_shape;
+  ref_shape.resize(in.num_samples());
+  int nfft = args.nfft > 0 ? args.nfft : args.window_length;
+  int spectrum_size = (nfft + 2) / 2;
+  for (int i = 0; i < in.num_samples(); i++) {
+    TensorShape<2> ts;
+    ts[t_axis] = args.num_windows(in.shape[i][0]);
+    ts[f_axis] = spectrum_size;
+    ref_shape.set_tensor_shape(i, ts);
+  }
+  ref_out.reshape(ref_shape);
+  RefSpectrum(ref_out.cpu(), in, args, window);
+}
+
+TEST(StftGPU, Setup) {
+  StftGPU stft;
+  TensorListShape<1> lengths = {{ 100, 1000, 13, 45 }};
+  StftArgs args;
+  args.axis = 0;
+  args.spectrum_type = FFT_SPECTRUM_COMPLEX;
+  args.window_length = 1024;
+  args.window_center = 512;
+  args.window_step = 512;
+  for (bool time_major : { false, true }) {
+    args.time_major_layout = time_major;
+
+    KernelContext ctx;
+    KernelRequirements req = stft.Setup(ctx, lengths, args);
+    ASSERT_EQ(req.output_shapes.size(), 1u);
+    auto &o_shape = req.output_shapes[0];
+    ASSERT_EQ(o_shape.num_samples(), 4);
+    for (int i = 0; i < 4; i++) {
+      TensorShape<2> ts = { args.num_windows(lengths.shapes[i]), args.window_length / 2 + 1 };
+      if (!time_major)
+        std::swap(ts[0], ts[1]);
+      EXPECT_EQ(o_shape[i], ts);
+    }
+  }
+}
+
+void GenerateTestSound(float *out, int length, float freq, float amplitude) {
```

Comment:
this could be useful in another tests. Maybe we should extracted to some common test utils

Reply:
Only if we like soundtracks from sci-fi movies from '50s :)